### ISSUE TITLE
Add CSS/HTML linting?

### ISSUE THREAD

**Issue by pdehaan** [mozilla/testpilot-containers#72]
Not sure how much value it adds, but out of curiousity I already ran the code through linters already.

Since we're using CSS variables, I couldn't seem to get **sass-lint**, **csslint** or **csslint-next** up working... but turns out [**stylelint**](http://npm.im/stylelint) works like a champ!

#### .stylelintrc
```yaml
extends:
  - stylelint-config-standard

ignoreFiles:
  - webextension/css/*.min.css

rules:
  declaration-block-no-duplicate-properties: true
  # declaration-block-properties-order: alphabetical
```

#### .htmllintrc
```json
{
  "plugins": [],

  "id-class-style": "dash",
  "indent-style": "spaces",
  "indent-width": 2
}
```

#### package.json

```js
  "devDependencies": {
    "addons-linter": "^0.15.14",
    "eslint": "^3.12.2",
    "htmllint-cli": "^0.0.5",
    "jpm": "^1.2.2",
    "stylelint": "^7.7.1",
    "stylelint-config-standard": "^15.0.1"
  },
  "scripts": {
    "lint": "npm-run-all lint:*",
    "lint:addon": "addons-linter webextension --self-hosted"
    "lint:css": "stylelint webextension/css/*.css",
    "lint:html": "htmllint webextension/*.html",
    "lint:js": "eslint ."
  }
```

---

### Current output:

```sh
$ stylelint webextension/css/*.css

webextension/css/popup.css
  2:15  ✖  Expected "#FAFAFA" to be "#fafafa"                               color-hex-case
 15:3   ✖  Unexpected empty line before declaration                         declaration-empty-line-before
 35:12  ✖  Expected a leading zero                                          number-leading-zero
 66:15  ✖  Expected "#DCDBDC" to be "#dcdbdc"                               color-hex-case
 71:12  ✖  Expected a leading zero                                          number-leading-zero
 90:25  ✖  Expected single space after ":" with a single-line declaration   declaration-colon-space-after
```

Although obviously everything is super-configurable, so we can tweak CSS/HTML rules for days to the place where we're all happy with the ruleset.

**Comment by johngruen** [comment]
classic @pdehaan issue

**Comment by jonathanKingston** [comment]
I feel in awe of someone with as much desire of linting as I have 👍.

Will add a PR for the suggested config and fixes.

**Comment by pdehaan** [comment]
Fixed in #110. Closing. 🍻
